docs/blog/observability-at-scale.md

```diff
@@ -0,0 +1,109 @@
+
+### **Pull vs. push**
+
+Pull and push modes relate to the direction of data flow. If the agent collects data and pushes them to the backend for further analysis, we call it "push" mode. Debate over pull vs. push has gone on for a long time. The key for an observability system is to minimize the cost of the agent, and to be generally suitable for different kinds of observability data.
+
+The agent would send the data out a short period after it is collected. Then, we would have less concern about overloading the local cache. One typical case would be endpoint (URI of HTTP, service of gRPC) metrics. Any service could easily have hundreds, even thousands of endpoints. An APM system must have these metrics analysis capabilities.
+
+Furthermore, metrics aren't the only thing in the observability landscape; traces and logs are important too. SkyWalking is designed to provide a 100% sampling rate tracing capability in the production environment. Clearly, push mode is the only solution.
+
+At the same time, using push mode natively doesn't mean SkyWalking can't do data pulling. In recent 8.x releases, SkyWalking supports fetching data from Prometheus-instrumented services for reducing the Non-Recurring Engineering of the end users. Also, pull mode is popular in the MQ based transport, typically as a Kafka consumer. The SkyWalking agent side uses the push mode, and the OAP server uses the pull mode.
+
+The conclusion: push mode is the native way, but pull mode works in some special cases too.
+
+### **Metrics analysis isn't just mathematical calculation**
+
+Metrics rely on mathematical theories and calculations. Percentile is a good measure for identifying the long tail issue, and reasonable average response time and successful rate are good SLO(s). But those are not all. Distributed tracing provides not just traces with detailed information, but high values metrics that can be analyzed.
+
+The service topology map is required from Ops and SRE teams for the NOC dashboard and confirmation of system data flow. SkyWalking uses the [STAM (Streaming Topology Analysis Method)](https://wu-sheng.github.io/STAM/) to analyze topology from the traces, or based on ALC (Envoy Access Log Service) in the service mesh environment. This topology and metrics of nodes (services) and lines (service relationships) can't be pulled from simple metrics SDKs.
+
+![SkyWalkingTopology](https://lh5.googleusercontent.com/mmEhxSqUQOzFPWNWNkGEzML0g9b72TgbKbNJexNe-Ok1jC66LUq-g5jdOQe3MKd_a0DT5fud6_NtqGdOSTus-y4rQ3aoBOp44wRmofN6IEnvegZy3sahOLghn37W55ybQWgyayVq)
```

Comment:
It is better to host the images in this repo, as Google is blocked in China :)

Reply:
I am probably doing something wrong but my attempt to add the images to the repo didn't work, I got a message push access is required to add a file to the repo at: https://github.com/apache/skywalking-website/upload/master/docs/.vuepress/public/static/blog

Reply:
Work on this repo, https://github.com/tevahp/skywalking-website/, branch `patch-1`.